File: Patterns/PatternsVasylyk/Prototype/Prototype/Student.cs
namespace Prototype;

    public class Student: Person
    {
        public Student(string name, Teacher teacher) : base(name)
        {
            Teacher = teacher;
        }

        public Teacher Teacher { get; set; }
        public override Person Clone()
        {
            Student studentClone = (Student)MemberwiseClone();
            studentClone.Teacher = new Teacher(this.Teacher.Name, this.Teacher.Course );
            return studentClone;
        }
    }
        
        
     

File: Patterns/Pasternak/Composite/Student.java
package Pattern.Composite;

public class Student implements Person{
    public String university;
    public Student(String university){
        this.university = university;
    }
    @Override
    public void print() {
        System.out.println("Я Студент і я навчаюсь в " + university);
    }
}
